application/F3DStarter.cxx

```diff
@@ -651,6 +659,88 @@ class F3DStarter::F3DInternals
     }
   }
 
+  // Update bindings by initializing them and adding the app and config bindings
+  void UpdateBindings(const std::vector<fs::path>& paths)
+  {
+    if (!this->AppOptions.NoRender)
+    {
+      // std::vector<std::string> -> std::string for printing
+      auto vecToString = [](const std::vector<std::string>& vec)
+      {
+        std::string ret;
+        for (const std::string& val : vec)
+        {
+          ret += "`";
+          ret += val;
+          ret += "` ";
+        }
+        return ret;
+      };
+
+      // "doc", ""
+      auto docString = [](const std::string& doc) { return std::make_pair(doc, ""); };
+
+      auto docStringVec = [&](const std::vector<std::string>& docs)
+      { return std::make_pair(vecToString(docs), ""); };
+
+      using mod_t = f3d::interaction_bind_t::ModifierKeys;
+
+      f3d::interactor& interactor = this->Engine->getInteractor();
+      interactor.initBindings();
+
+      // clang-format off
+      interactor.addBinding({ mod_t::NONE, "Left" }, "load_previous_file_group", "Others", std::bind(docString, "Load previous file group"));
+      interactor.addBinding({ mod_t::NONE, "Right" }, "load_next_file_group", "Others", std::bind(docString, "Load next file group"));
+      interactor.addBinding({ mod_t::NONE, "Up" }, "reload_current_file_group", "Others", std::bind(docString, "Reload current file group"));
+      interactor.addBinding({ mod_t::NONE, "Down" }, "add_current_directories", "Others", std::bind(docString, "Add files from dir of current file"));
+      interactor.addBinding({ mod_t::NONE, "F11" }, "take_minimal_screenshot", "Others", std::bind(docString, "Take a minimal screenshot"));
+      interactor.addBinding({ mod_t::NONE, "F12" }, "take_screenshot", "Others", std::bind(docString, "Take a screenshot"));
+
+      // This replace an existing default binding command in the libf3d
+      interactor.removeBinding({ mod_t::NONE, "Drop" });
+      interactor.addBinding({ mod_t::NONE, "Drop" }, "add_files_or_set_hdri", "Others", std::bind(docString, "Load dropped files, folder or HDRI"));
+      interactor.addBinding({ mod_t::CTRL, "Drop" }, "add_files", "Others", std::bind(docString, "Load dropped files or folder"));
+      interactor.addBinding({ mod_t::SHIFT, "Drop" }, "set_hdri", "Others", std::bind(docString, "Set HDRI and use it"));
+      // clang-format on
+
+      f3d::log::debug("Adding config defined bindings if any: ");
+      bool logBindings = this->AppOptions.VerboseLevel == "debug";
+      std::map<std::string, std::tuple<std::string, fs::path, std::string, std::string>> loggingMap;
```

Comment:
Any reason you are ordering by alphabetical order? Don't you want to print bindings by config file order?

Reply:
I want to print the actual result so I need a map for unicity.
Having unicity + order would be nice but since this is only a log it doesnt really matter.

Reply:
ok